torch/distributed/rpc/server_process_global_profiler.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/python3
+
+import torch
+from torch.autograd import ProfilerConfig
+
+from . import (
+    _disable_server_process_global_profiler,
+    _enable_server_process_global_profiler,
+)
+
+
+def enable_server_process_global_profiler(config: ProfilerConfig):
+    r"""
+    Turn on the state that indicates the server-side process-global profiling is
+    on. This enables all RPC threads running server-side request callbacks.
+
+    Arguments:
+        config (torch.autograd.profiler.ProfilerConfig): This config specifies
+            1) what mode. CPU only or CPU + CUDA.
+            2) Whether to record input shapes.
+
+    Returns:
+        None
+
+    """
+    _enable_server_process_global_profiler(config)
+
+
+def disable_server_process_global_profiler():
```

Comment:
IIUC, the current version could return more events than those that are issued by the current worker? And the this worker needs to somehow filter those events to get relevant ones? If so, shall we make these two APIs private for now (starts with "_" and also need to address name conflict)? Because it doesn't seem very user-friendly. We might need more helpers/examples/tutorials for these APIs. If the purpose for now is jus to unblock internal use cases, private functions should be sufficient?

Reply:
We don't need these helpers, we only care about averages of all events.

The worker (usually a master) that enable this global profiler does not care about which worker put works on the server. It wants to get all profiling events on that server no matter which worker initiates the work.

Reply:
By "we" do you mean the current internal use case, or all future possible use cases including OSS?

Reply:
Current internal use case.